backend/src/scripts/utility/scene/getMeshProperties.js

```diff
@@ -0,0 +1,127 @@
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This script is used to extract information aboute the geometry in
+ * unoptimised scenes.
+ */
+
+const { v5Path } = require('../../../interop');
+
+const { logger } = require(`${v5Path}/utils/logger`);
+const { getTeamspaceList, getCollectionsEndsWith } = require('../../utils');
+const Path = require('path');
+const fs = require('fs');
+
+const { find } = require(`${v5Path}/handler/db`);
+
+// In the resulting file, identify each revision by a simple integer
+let globalSceneCounter = 0;
+let globalMeshCounter = 0;
+let globalFs = null;
+
+const printUpdate = () => {
+	logger.logInfo(`Processed ${globalMeshCounter} meshes across ${globalSceneCounter} revisions...`);
+};
+
+const processMeshNodes = (meshNodes) => new Promise((resolve) => {
+	const buffer = Buffer.alloc(4 * 5 * meshNodes.length);
+	let i = 0;
+	for (const node of meshNodes) {
+		buffer.writeUint32LE(globalSceneCounter, (i++) * 4);
+		const x = node.bounding_box[1][0] - node.bounding_box[0][0];
+		const y = node.bounding_box[1][1] - node.bounding_box[0][1];
+		const z = node.bounding_box[1][2] - node.bounding_box[0][2];
+		buffer.writeFloatLE(x, (i++) * 4);
+		buffer.writeFloatLE(y, (i++) * 4);
+		buffer.writeFloatLE(z, (i++) * 4);
+		buffer.writeUint32LE(node.vertices_count, (i++) * 4);
+	}
+	globalMeshCounter += meshNodes.length;
+	if (!globalFs.write(buffer)) {
+		globalFs.once('drain', resolve);
+	} else {
+		resolve();
+	}
+});
+
+const processScene = async (teamspace, name, rev) => {
+	const query = {
+		type: { $eq: 'mesh' },
+		rev_id: { $eq: rev._id },
+	};
+	const properties = {
+		vertices_count: 1,
+		faces_count: 1,
+		bounding_box: 1,
+		primitive: 1,
+	};
+	const meshNodes = await find(teamspace, `${name}.scene`, query, properties);
+	if (meshNodes.length) {
+		await processMeshNodes(meshNodes);
+	}
+	globalSceneCounter++;
+};
+
+const processProject = async (teamspace, name) => {
+	const revisions = await find(teamspace, `${name}.history`);
+	for (const rev of revisions) {
+		// eslint-disable-next-line no-await-in-loop
+		await processScene(teamspace, name, rev);
+	}
+};
+
+const processTeamspace = async (teamspace) => {
+	const cols = await getCollectionsEndsWith(teamspace, '.history');
+	for (const { name } of cols) {
+		// eslint-disable-next-line no-await-in-loop
+		await processProject(teamspace, name.slice(0, -('.history'.length)));
+	}
+};
+
+const run = async (filename) => {
+	logger.logInfo('Creating output file...');
+	globalFs = fs.createWriteStream(filename, {
+		autoClose: true,
+		encoding: 'binary',
+	});
+	setInterval(printUpdate, 3000);
+	logger.logInfo('Finding all teamspaces...');
+	const teamspaces = await getTeamspaceList();
+	for (const teamspace of teamspaces) {
+		// eslint-disable-next-line no-await-in-loop
```

Comment:
would be useful to echo which teamspace is being worked on for logging purposes but not a big deal.